policy/iagent.go

```diff
@@ -0,0 +1,21 @@
+package policy
+
+import (
+	"context"
+
+	"github.com/veraison/services/config"
+	"github.com/veraison/services/proto"
+)
+
+type IAgent interface {
+	SetBackEnd(b IBackend)
+	GetBackEnd() IBackend
```

Comment:
inconsistent camelcase (should be `GetBackend`, `SetBackend`)

but more importantly: why are these two methods needed?  I thought the agent is created once and for all from config, whereas here we'd allow callers to swap it on the fly.

Reply:
To your question in relevance of SetBackend: It would be a nice to have feature to allow callers to have a control about what BackEnd they want to use ? I am not certainly using this feature as of now, but can keep it if we agree!

For GetBackend, we are actually using it to check the Backend in the tests and now it been an interface, I do not see any wrong in keeping this so would like to retain this...

Reply:
> To your question in relevance of SetBackend: It would be a nice to have feature to allow callers to have a control about what BackEnd they want to use ? I am not certainly using this feature as of now, but can keep it if we agree!

I think the intended API contract is to create the object from config and then use it with the understanding that it won't change midlife.
Therefore, I don't think setter is needed (or even wanted).

> For GetBackend, we are actually using it to check the Backend in the tests and now it been an interface, I do not see any wrong in keeping this so would like to retain this...

If it is only used in tests, then it's not necessary to put it in the public interface: since tests are in the same package as the implementation, they can access the object attributes directly.

In summary, I suggest we drop both setter and getter.

Reply:
We cannot check from an Interface, which Backend is been used without this method in the test?

Reply:
ah right, the Create does not return a concrete type that can be inspected directly, OK!  Still it doesn't feel right to expose an interface method only for testing reasons.

Reply:
The change is inline with the current API GetBackendname().

I see no issue with allowing external entities to query what is set from Config, which is Name of the Backend and Backend itself??